src/server/configuration.js

```diff
@@ -238,3 +241,9 @@ export const amqpUri = config.get("amqp.amqp_uri");
  *
  */
 export const amqpExchangeName = config.get("amqp.exchange_name");
+
+/**
+ * Is irods home path
+ * @type {string}
+ */
+export const irodsHomePath = config.get("irods.home_path");
```

Comment:
I wonder if we really need this config at all. Right now, `bootstrap` returns a `data_info` property 

` data_info: {
        user_home_path: "/iplant/home/jmittelmeier",
        user_trash_path: "/iplant/trash/home/de-irods/jmittelmeier",
        base_trash_path: "/iplant/trash/home/de-irods",
    },`

Can we simply add irods home path to this?

Reply:
This will be required when the `data-info` service is down, or maybe if `bootstrap` hasn't responded yet. We use it in the current UI this way:
https://github.com/cyverse-de/ui/blob/58224562a7b44d4900cea6b5b195a5bd1fab128c/de-lib/src/main/java/org/iplantc/de/client/models/UserInfo.java#L201-L216

So it appears the `launch` page is now imitating the logic of that `UserInfo` model, and constructing a reasonable assumption of the user's home folder, but it's only using it to construct an assumption of the user's default output folder.

Ultimately the user's default output folder should come from their preferences, and their home folder will come from the `bootstrap` response, but whatever will be responsible for storing those preferences or bootstrap values might also need this config to manually create a default home folder in case a service can't be reached?

Reply:
Sounds good. Can we launch a job when data-info is down ? I was under the assumption we could not.

I agree - users default folder should come from user preferences / bootstrap. I know my PR for preferences is not ready yet but, I am prefetching and caching bootstrap info. In case of failure, there will be retries as well. So if we don't have bootstrap info in our code, then bootstrap service is really down. And in that case, can `apps/analyses` service get the users home folder / default output folder from `data-info `. Does `launch` service expects the output folder set when launching a job or can it be empty?

Reply:
We can move this to a design meeting. I have few more thoughts / questions about apps, app launch and analyses.

Reply:
We can move this to a design meeting, but a few points to keep in mind for the meeting:

I wasn't intending on keeping this logic of building a default output folder in this page, it's just here until it can be grabbed from the `bootstrap` code. I was defending keeping this config, though, since this config and the logic that uses it was added as part of the UI resiliency epic (cyverse-de/ui#50).

> Sounds good. Can we launch a job when data-info is down ? I was under the assumption we could not.
> 

No, I don't think we can, but I think there's a possibility that `data-info` doesn't respond for the `bootstrap` call, but is available again once the user is ready to launch the analysis.

> So if we don't have bootstrap info in our code, then bootstrap service is really down. And in that case, can `apps/analyses` service get the users home folder / default output folder from `data-info `? Does `launch` service expects the output folder set when launching a job or can it be empty?

Well the `bootstrap` endpoint is in `terrain`, so if the entire `bootstrap` endpoint is down then nothing works 😉 
However, the `bootstrap` endpoint gets the path info from `data-info`, but it can still get as much info as possible if `data-info` is down, and only in that case will the home and trash paths be missing for the UI.

Finally, there is a known issue with an analysis output folder that I think is also an issue in the current UI. If the default output folder is not available when the launch form is created, and the user picks an output folder, then this form will assume that the user wants to use that exact folder for outputs, and the job won't automatically create a subfolder under it:
https://github.com/cyverse-de/sonora/blob/0c60284dba2e9e3c328e6db4b25b09776243f594/src/components/apps/launch/AppLaunchForm.js#L211

This means if there is already a file under that folder that has the same name as an output file, it will be overwritten by this job.

Again, this form is imitating the logic of the current UI, but we can add a ticket for this issue and fix it in another PR if necessary (I don't know the history behind the decision to hide the `create_output_subdir` flag from the user).

Reply:
Thanks for detailed response. I guess what have I in my mind is new approach. I will try elaborate more on this in the meeting.